_docs/developer/testing/linting_static_analysis.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Linting / Static Analysis
+---
+
+To help ensure the [Coding Standards](/developer/coding_style_guide) of Submitty, we use a mixture
+of linting and static analysis. A linter is per [wikipedia](https://en.wikipedia.org/wiki/Lint_(software))
+is "tool that analyzes source code to flag programming errors, bugs, stylistic errors, and suspicious constructs."
+Through the linter, we ensure a consistent coding style as you go from one file to the next within a given language,
+regardless of portion of the code. Static Analysis, similar to a linter, analyzes the source code, but in this
+case constructs abstract syntax trees to validate behavior of the code such as using the right number of types
+of parameters, functions return what they say they do, etc.
+
+## Python Linting
+
+The Python code of Submitty is linted using [flake8](https://flake8.pycqa.org/en/latest/) and 
+[flake8-bugbear](https://github.com/PyCQA/flake8-bugbear). You can run it at the root level of
+Submitty to test all Python files by just doing:
+
+```bash
+flake8
```

Comment:
should we replace this command with (or add) "`python3 -m flake8` on Linux/Mac" ? Unless there's a config step I missed when installing flake8, I haven't been able to use it like that.

Reply:
Sounds like you've not added python's packages' bin folder to your path and so `flake8` is not discoverable.

Changed this to `python3 -m flake8` though since it's guaranteed to work as you point out.